backend/src/modules/idim-webservice/idim-webservice.service.ts

```diff
@@ -255,4 +256,113 @@ export class IdimWebserviceService {
+                    requestData,
+                    function (error, foundUser) {
+                        // this will be any error from the IDIM server side
+                        if (error) {
+                            return reject(
+                                new HttpException(
+                                    {
+                                        error:
+                                            'IDIM web service call error: ' +
+                                            error,
+                                    },
+                                    HttpStatus.INTERNAL_SERVER_ERROR
+                                )
+                            );
+                        }
+
+                        // this will be any error return by the web service
+                        // for example if we provided an non existing requestor id, or permission issue
+                        if (
+                            foundUser.getAccountDetailResult.code == 'Failed' &&
+                            foundUser.getAccountDetailResult.failureCode !==
+                                'NoResults'
+                        ) {
+                            return reject(
+                                new HttpException(
+                                    {
+                                        status: HttpStatus.BAD_REQUEST,
+                                        code: foundUser.getAccountDetailResult
+                                            .code,
+                                        failureCode:
+                                            foundUser.getAccountDetailResult
+                                                .failureCode,
+                                        message:
+                                            foundUser.getAccountDetailResult
+                                                .message,
+                                    },
+                                    HttpStatus.BAD_REQUEST
+                                )
+                            );
+                        }
+
+                        // user not found case
+                        // getAccountDetail method returns code Failed with failureCode NoResults
+                        // which is different than the not found case of searchInternalAccount method that we used above for searching idir user
+                        if (
+                            foundUser.getAccountDetailResult.code == 'Failed' &&
+                            foundUser.getAccountDetailResult.failureCode ==
+                                'NoResults'
+                        ) {
+                            const response = new BCEIDUserResponse();
+                            response.found = false;
+                            if (
+                                searchUserBy == SearchUserParameterType.UserGuid
+                            )
+                                response.guid = searchValue;
```

Comment:
this method is almost the same as the old one, only different is in the not found case, depends on the "searchUserBy" parameter returns found false with userId or guid
<img width="295" alt="Screen Shot 2024-04-29 at 4 47 10 PM" src="https://github.com/bcgov/nr-fam-idim-lookup-proxy/assets/23131735/4c7e931a-d09b-48a2-abe3-9973e8933ad3">
<img width="314" alt="Screen Shot 2024-04-29 at 4 47 19 PM" src="https://github.com/bcgov/nr-fam-idim-lookup-proxy/assets/23131735/e97dd8e5-2ed1-4b9d-8d15-97f59f14547c">